canal/sync.go

```diff
@@ -96,12 +96,17 @@ func (c *Canal) runSyncBinlog() error {
 			// we only focus row based event
 			err = c.handleRowsEvent(ev)
 			if err != nil {
-				e := errors.Cause(err)
-				// if error is not ErrExcludedTable or ErrTableNotExist or ErrMissingTableMeta, stop canal
-				if e != ErrExcludedTable &&
-					e != schema.ErrTableNotExist &&
```

Comment:
in the old code, the error may come from any `return` branch in `handleRowsEvent`, but in your change we only checked the last `return`. I made a quick check that the three errors of

```
		if e != ErrExcludedTable &&
			e != schema.ErrTableNotExist &&
			e != schema.ErrMissingTableMeta {
```

seems only come from `c.GetTable`

Reply:
I agree with you. In this case, I think the old code means to ignore three errors: ErrExcludedTable,schema.ErrTableNotExist, schema.ErrMissingTableMeta.  So we can do this just after GetTable in handleRowsEvent